conan/tools/meson/toolchain.py

```diff
@@ -3,187 +3,159 @@
         self._conanfile = conanfile
-        self._backend = self._get_backend(backend)
-        self._build_type = self._conanfile.settings.get_safe("build_type")
-        self._base_compiler = self._conanfile.settings.get_safe("compiler.base") or \
-                              self._conanfile.settings.get_safe("compiler")
-        self._vscrt = self._conanfile.settings.get_safe("compiler.base.runtime") or \
-                      self._conanfile.settings.get_safe("compiler.runtime")
-        self._cppstd = cppstd_from_settings(self._conanfile.settings)
-        self._shared = self._conanfile.options.get_safe("shared")
-        self._fpic = self._conanfile.options.get_safe("fPIC")
-        self._build_env = VirtualBuildEnv(self._conanfile).vars()
-
-        self.definitions = dict()
-        self.preprocessor_definitions = dict()
-
-        def from_build_env(name):
-            return self._to_meson_value(self._build_env.get(name, None))
+        # Values are kept as Python built-ins so users can modify them more easily, and they are
+        # only converted to Meson file syntax for rendering
+        # priority: first user conf, then recipe, last one is default "ninja"
+        backend_conf = conanfile.conf["tools.meson.mesontoolchain:backend"]
+        self.backend = backend_conf or backend or 'ninja'
 
-        self.c = from_build_env("CC")
-        self.cpp = from_build_env("CXX")
-        self.c_ld = from_build_env("CC_LD") or from_build_env("LD")
-        self.cpp_ld = from_build_env("CXX_LD") or from_build_env("LD")
-        self.ar = from_build_env("AR")
-        self.strip = from_build_env("STRIP")
-        self.as_ = from_build_env("AS")
-        self.windres = from_build_env("WINDRES")
-        self.pkgconfig = from_build_env("PKG_CONFIG")
+        build_type = self._conanfile.settings.get_safe("build_type")
+        self.buildtype = {"Debug": "debug",  # Note, it is not "'debug'"
+                          "Release": "release",
+                          "MinSizeRel": "minsize",
+                          "RelWithDebInfo": "debugoptimized"}.get(build_type, build_type)
+        self.b_ndebug = "true" if self.buildtype != "Debug" else "false"
 
+        # https://mesonbuild.com/Builtin-options.html#base-options
+        fpic = self._conanfile.options.get_safe("fPIC")
+        shared = self._conanfile.options.get_safe("shared")
+        self.b_staticpic = fpic if (shared is False and fpic is not None) else None
         # https://mesonbuild.com/Builtin-options.html#core-options
         # Do not adjust "debug" if already adjusted "buildtype"
-        self.buildtype = self._to_meson_build_type(self._build_type) if self._build_type else None
-        self.default_library = self._to_meson_shared(self._shared) \
-            if self._shared is not None else None
-        self.backend = self._to_meson_value(self._backend)
+        self.default_library = ("shared" if shared else "static") if shared is not None else None
```

Comment:
I don't think we should ever expose all these variables to the public.
they are not supposed to be assigned to the random values from the outside.
they only gain limited set of the values computed from the settings as the input.

Reply:
Made them private. At some point we will want to make it more flexible and customizable, but i agree on hiding them atm